### PR TITLE
[networks] Enforce concurrency limit

### PR DESCRIPTION
**Pull request by p-lambert**
### What does this PR do?

Enforces a maximum number of concurrent requests for certain system-probe endpoints.

### Motivation

Ensure that in-flight requests won't pile up in the event of a system-probe bug.

### Describe how to test/QA your changes

Change covered by unit tests.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/system-probe/api/server.go [hunk 1]
```diff
@@ -33,10 +33,10 @@ func StartServer(cfg *config.Config) error {
 	}
 
 	// Register stats endpoint
-	mux.HandleFunc("/debug/stats", func(w http.ResponseWriter, req *http.Request) {
+	mux.HandleFunc("/debug/stats", utils.WithConcurrencyLimit(5, func(w http.ResponseWriter, req *http.Request) {
```
Comment (derekwbrown) [inline cmd/system-probe/api/server.go:36]: At minimum, a const at the top instead of magic number?
Also, should this have a configurable?

**Review by derekwbrown — APPROVED** [review]
Overall seems fine modulo the one comment.
Feel free to check in afterward...

File: cmd/system-probe/api/server.go (hunk 1 above)
Comment (p-lambert) [reply cmd/system-probe/api/server.go:36]: Good point – I added a const and a comment

Comment (derekwbrown) [reply cmd/system-probe/api/server.go:36]: looks like you didn't use the const here though.

**Pull request merged**